### PR TITLE
Type inference

### PR DESCRIPTION
**Pull request by maylim17**

### CODE REVIEW

File: src/inferencer/inferencer.ts [hunk 1]
```diff
@@ -1,37 +1,46 @@
...
+  }
+
   function inferLiteral(literal: TypeAnnotatedNode<es.Literal>) {
     const valueOfLiteral = literal.value
     if (typeof valueOfLiteral === 'number') {
       // declare
       literal.inferredType = {
         kind: 'primitive',
-        name: 'number'
+        name: 'integer'
```
Comment (C-likethis123) [inline src/inferencer/inferencer.ts]: Why is this changed to 'integer'?

File: src/inferencer/inferencer.ts [hunk 2]
```diff
@@ -1,37 +1,46 @@
 import { ancestor } from 'acorn-walk/dist/walk'
-import { TypeAnnotatedNode } from '../types'
+import { TypeAnnotatedNode, Variable } from '../types'
 import { annotateProgram } from './annotator'
+import { primitiveMap } from './typeEnvironment'
 import * as es from 'estree'
 
 // // main function that will infer a program
 export function inferProgram(program: es.Program): TypeAnnotatedNode<es.Program> {
+  const typeConstraints = new Map()
```
Comment (C-likethis123) [inline src/inferencer/inferencer.ts]: Might be a good idea to have another file for type constraint related logic, so we don't clutter this file.

File: src/inferencer/inferencer.ts (hunk 1 above)
Comment (maylim17) [reply src/inferencer/inferencer.ts]: I will change back to number
